server/app/index.js

```diff
@@ -1,85 +1,84 @@
-import dotenv from "dotenv";
-dotenv.config();
-import {
-  NODE_ENV,
-  DEFAULT_PORT,
-  DEFAULT_ENDPOINT,
-  DEFAULT_APOLLO_INTROSPECTION,
-  DEFAULT_APOLLO_PLAYGROUND
-} from "./config/default.js";
-import { ApolloServer } from "apollo-server-express";
-import { ApolloServerPluginDrainHttpServer } from "apollo-server-core";
+import "./config/env.js";
 import express from "express";
 import http from "http";
-import path from "path";
-const __dirname = path.resolve();
-import { readFile } from "fs/promises";
-const pkg = JSON.parse(
-  await readFile(new URL("./package.json", import.meta.url))
-);
+import { ApolloServer } from "apollo-server-express";
+import { ApolloServerPluginDrainHttpServer } from "apollo-server-core";
 // Database
 import { initDatabase } from "./config/database.js";
+// View engine
+import viewEngine from "./config/viewEngine.js";
+// Routes
+import homeRouter from "./routes/home.js";
 // GraphQL
+import { makeExecutableSchema } from "@graphql-tools/schema";
+import { useSofa } from "sofa-api";
+// Definitions
 import { typeDefs } from "./graphql/types.js";
 import { resolvers } from "./graphql/resolvers.js";
+const schema = makeExecutableSchema({
+  typeDefs,
+  resolvers
+});
 
-async function startServer(typeDefs, resolvers) {
+/**
+ * Main function to init and start server
+ */
+async function startServer() {
+  // Express server
   const app = express();
+  // const router = express.Router();
+  // Used by ApolloServerPluginDrainHttpServer - See https://www.apollographql.com/docs/apollo-server/api/plugin/drain-http-server/
   const httpServer = http.createServer(app);
+
+  // Apollo server
   const server = new ApolloServer({
-    typeDefs,
-    resolvers,
+    schema,
+    formatError: (error) => ({
+      name: error.name,
+      message: error.message.replace("Error", "")
+    }),
     plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
-    introspection:
-      process.env.APOLLO_INTROSPECTION || DEFAULT_APOLLO_INTROSPECTION,
-    playground: process.env.APOLLO_PLAYGROUND || DEFAULT_APOLLO_PLAYGROUND,
-    path: process.env.ENDPOINT || DEFAULT_ENDPOINT
+    introspection: process.env.APOLLO_INTROSPECTION,
+    playground: process.env.APOLLO_PLAYGROUND,
+    path: process.env.GRAPHQL_ENDPOINT
   });
 
-  await server.start();
-
-  // View engine setup
-  app.set("views", path.join(__dirname, "./views"));
-  app.set("view engine", "pug");
-  app.use(express.static(path.join(__dirname, "./public")));
+  // Init view engine
+  viewEngine(app);
 
-  function creditYears() {
-    const currentYear = new Date().getFullYear();
-    if (currentYear > 2022) return `2022-${currentYear}`;
-    else return currentYear;
-  }
+  // Define routes
+  // Server landing page
+  app.use("/", homeRouter);
+  // REST API
+  app.use(
+    process.env.REST_BASE,
+    useSofa({
```

Comment:
would be interesting to also use the sofa feature to auto generate the Open API documentation

Reply:
Pull the branch and look at my last commit.
The doc will be served through /{REST_BASE}/docs (http://localhost:4000/rest/docs/ by default)